server/src/main/java/org/opensearch/index/store/RemoteStoreFileTrackerAdapter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store;
+
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.index.store.remote.utils.BlobFetchRequest;
+import org.opensearch.index.store.remote.utils.filetracker.FileState;
+import org.opensearch.index.store.remote.utils.filetracker.FileType;
+
+import java.io.IOException;
+
+public interface RemoteStoreFileTrackerAdapter {
+    IndexInput fetchBlob(BlobFetchRequest blobFetchRequest) throws IOException;
+
+    void trackFile(String name, FileState fileState, FileType fileType);
+
+    void updateFileType(String name, FileType fileType);
+
+    void updateFileState(String name, FileState fileState);
+
+    void removeFileFromTracker(String name);
+
+    FileState getFileState(String name);
+
+    FileType getFileType(String name);
+
+    boolean isFilePresent(String name);
+}
```

Comment:
Can you add a Java doc explaining what a tracker "adapter" is supposed to do. 
My view point is consumers of tracker shouldn't directly be modifying low level stuffs like state, type etc.

Reply:
Based on discussions in the [RFC](https://github.com/opensearch-project/OpenSearch/issues/12781), have decided to move away from the FileTracker and RemoteStoreFileTrackerAdapter and directly injecting RemoteDirectory in CompositeDirectory